astronomer/providers/google/cloud/extractors/bigquery_async_extractor.py

```diff
@@ -0,0 +1,85 @@
+from typing import Any, List, Optional
+
+from airflow.exceptions import AirflowException
+from airflow.models.taskinstance import TaskInstance
+from airflow.utils.log.logging_mixin import LoggingMixin
+from google.cloud.bigquery import Client
+from openlineage.airflow.extractors.base import BaseExtractor, TaskMetadata
+from openlineage.airflow.utils import get_job_name
+from openlineage.common.provider.bigquery import BigQueryDatasetsProvider
+
+from astronomer.providers.google.cloud.operators.bigquery import (
+    BigQueryInsertJobOperatorAsync,
+)
+
+
+class BigQueryAsyncExtractor(BaseExtractor, LoggingMixin):
+    """
+    This extractor provides visibility on the metadata of a BigQuery Insert Job
+    including ``billedBytes``, ``rowCount``, ``size``, etc. submitted from a
+    ``BigQueryInsertJobOperatorAsync`` operator.
+    """
+
+    def __init__(self, operator: BigQueryInsertJobOperatorAsync):
+        super().__init__(operator)
+
+    def _get_big_query_client(self) -> Client:
+        """
+        Gets the BigQuery client to fetch job metadata.
+        The method checks whether a connection hook is available with the Airflow configuration for the operator, and
+        if yes, returns the same connection. Otherwise, returns the Client instance of``google.cloud.bigquery``.
+        """
+        hook = self.operator.hook
+        return hook.get_client(project_id=hook.project_id, location=hook.location)
+
+    def _get_xcom_bigquery_job_id(self, task_instance: TaskInstance) -> Any:
```

Comment:
Can we keep return type `str`?

Reply:
The method return the ouput of `xcom_pull` method which declares the return type to `Any`. I had kept the return type to `str` as in our case the jobID will be string, but mypy complained that I have to keep it aligned with the method's return type and make it to `Any`